fix: use --track-unscheduled-pods to select unscheduled pods in Daemonset sharding (#2388)
kubernetes/kube-state-metrics: docs/developer/cli-arguments.md

      --custom-resource-state-config-file string   Path to a Custom Resource State Metrics config file (experimental)
      --custom-resource-state-only                 Only provide Custom Resource State metrics (experimental)
      --enable-gzip-encoding                       Gzip responses when requested by clients via 'Accept-Encoding: gzip' header.

>       --enable-no-node-scrape                      This configuration is used in conjunction with node configuration. When this configuration is true, node configuration is empty and the metric of no scheduled pods is scraped. This is experimental.

Reviewer: Suggestion: Move this closer to the `--node` argument and rename from no_node to unscheduled_pods or similar?

```suggestion
      --node-sharding-unscheduled-pods-only                     This configuration is used in conjunction with node configuration. When this configuration is true, node configuration is empty and the metric of unscheduled pods is scraped from the Kubernetes API. This is experimental.
```

>       --fetch-unscheduled-pods                     This configuration is used in conjunction with node configuration. When this configuration is true, node configuration is empty and the metric of no scheduled pods is scraped. This is experimental.

Reviewer: Should the description make it more clear that this is only useful if you're doing sharding?

>       --enable-unscheduled-pods-fetching           This configuration is used in conjunction with node configuration. When this configuration is true, node configuration is empty and the metric of no scheduled pods is scraped. This is experimental.

Reviewer: ```suggestion
      --enable-unscheduled-pods-fetching           This configuration is used in conjunction with node configuration. When this configuration is true, node configuration is empty and the metric of unscheduled pods is fetched from the Kubernetes API Server. This is experimental.
```

>       --track-unscheduled-pods           This configuration is used in conjunction with node configuration. When this configuration is true, node configuration is empty and the metric of unscheduled pods is fetched from the Kubernetes API Server. This is experimental.

Reviewer 2: format
